src/storages/request_queue.js

```diff
@@ -312,9 +313,9 @@ export class RequestQueue {
         //    will try to fetch this request again, until it eventually appears in the main table.
         if (!request) {
             this.log.debug('Cannot find a request from the beginning of queue, will be retried later', { nextRequestId });
-            setTimeout(() => {
-                this.inProgress.delete(nextRequestId);
-            }, STORAGE_CONSISTENCY_DELAY_MILLIS);
+            await setTimeout(STORAGE_CONSISTENCY_DELAY_MILLIS);
```

Comment:
I don't think I've ever seen this message, which means it's probably not happening very often so I think it's ok to wait 3 seconds here.